### PR TITLE
Streamline SQL server docs

### PR DESCRIPTION
**Pull request by hadley**

### CODE REVIEW

File: R/db.R [hunk 1]
```diff
@@ -279,107 +279,97 @@ setMethod("sqlCreateTable", "DB2/AIX64",
...
 #'
+#' `conn@quote` returns the quotation mark, but quotation marks and square
+#' brackets can be interchangeably for delimited identifiers.
+#' (<https://learn.microsoft.com/en-us/sql/relational-databases/databases/database-identifiers>).
+#' So this function strips the brackets first, and then calls the DBI method to
+#' strip the quotation marks.
 #' @rdname SQLServer
 #' @docType methods
-#' @inheritParams DBI::dbUnquoteIdentifier
 #' @usage NULL
```
Comment (hadley) [inline R/db.R:300]: We'll be able to remove `@usage NULL` once dev roxygen2 is out. I made issue #666 (👿) to remember.

File: R/db.R [hunk 2]
```diff
@@ -279,107 +279,97 @@ setMethod("sqlCreateTable", "DB2/AIX64",
...
-#' \url{https://learn.microsoft.com/en-us/sql/relational-databases/databases/database-identifiers?view=sql-server-ver16}
-#' Therefore strip the brackets first, and then call the DBI method that strips
-#' the quotation marks.
-#' TODO: the generic implementation in DBI should take a quote char as
-#' parameter.
+#' @description
+#' ## `dbUnquoteIdentifier()`
 #'
+#' `conn@quote` returns the quotation mark, but quotation marks and square
+#' brackets can be interchangeably for delimited identifiers.
```
Comment (simonpcouch) [inline R/db.R:294]:
```suggestion
#' brackets can be used interchangeably for delimited identifiers.
```

File: R/db.R [hunk 3]
```diff
@@ -279,107 +279,97 @@ setMethod("sqlCreateTable", "DB2/AIX64",
...
+#' (<https://learn.microsoft.com/en-us/sql/relational-databases/databases/database-identifiers>).
+#' So this function strips the brackets first, and then calls the DBI method to
```
Comment (simonpcouch) [inline R/db.R:296]:
```suggestion
#' This function strips the brackets first and then calls the DBI method to
```

File: R/db.R (hunk 1 above)
Comment (simonpcouch) [reply R/db.R:300]: 😈👹👿

File: R/db.R [hunk 4]
```diff
@@ -279,107 +279,97 @@ setMethod("sqlCreateTable", "DB2/AIX64",
...
+        length(catalog_name) > 0 &&
+        catalog_name != "tempdb") {
       return(FALSE)
     }
 
-    if ( !grepl("^[#][^#]", name ) ) {
-      return(FALSE)
-    }
-    return(TRUE)
+    grepl("^[#][^#]", name)
```
Comment (simonpcouch) [inline R/db.R:328]: SATISfying

File: R/db.R [hunk 5]
```diff
@@ -279,107 +279,97 @@ setMethod("sqlCreateTable", "DB2/AIX64",
 
 # Microsoft SQL Server ---------------------------------------------------------
 
+#' SQL Server
+#'
+#' Details of SQL Server specific variations for odbc and DBI generics.
```
Comment (simonpcouch) [inline R/db.R:284]:
```suggestion
#' Details of SQL Server methods for odbc and DBI generics.
```

If this is technically accurate, it feels clearer to me.

File: R/db.R [hunk 6]
```diff
@@ -279,107 +279,97 @@ setMethod("sqlCreateTable", "DB2/AIX64",
...
+#' ## `isTempTable()`
 #'
 #' Local temp tables are stored as
-#' \code{ [tempdb].[dbo].[#name]________(padding using underscores)[numeric identifier] }
-#'
-#' True if:
-#' - If catalog_name is supplied it must equal "temdb" or "%" ( wildcard )
-#' - Name must start with "#" followd by a non-"#" character
+#' `[tempdb].[dbo].[#name]_____[numeric identifier]`, so `isTempTable()`
+#' returns `TRUE` if `catalog_name` is `"temdb"` or `"%"`, or the
```
Comment (simonpcouch) [inline R/db.R:315]:
```suggestion
#' returns `TRUE` if `catalog_name` is `"tempdb"` or `"%"`, or the
```

**Review by simonpcouch — APPROVED** [review]
Much clearer!

**Pull request merged**